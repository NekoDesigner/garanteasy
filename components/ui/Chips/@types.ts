export type TChipsIcon =
  | "hoursehold-electricals"
  | "small-electricals"
  | "diy"
  | "garden"
  | "fashion"
  | "other";

export interface IChipsProps {
  category: TChipsIcon | React.ReactNode;
  text: string;
  size?: "s" | "xs";
  showIcon?: boolean;
  onPress?: () => void;
}
