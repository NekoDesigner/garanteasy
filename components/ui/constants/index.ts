const SIZES = {
  icon: {
    s: 20,
    xs: 16,
  },
  padding: {
    xxs: 4,
    xs: 8,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 40,
    xxxl: 48,
  },
  font: {
    familly: {
      default: "Ubuntu" as const,
    },
    weight: {
      semiBold: "700" as const,
    },
    m: 16,
  },
};

const COLORS = {
  primary: "#1A1770",
  emerald: "#046C63",
  danger: "#C82828",
  light: "#FFFFFF",
  grey: "#EBEBEB",
  blueDarker: "#050426",
};

export { SIZES, COLORS };
