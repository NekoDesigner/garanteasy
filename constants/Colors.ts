export const COLORS = {
  primary: "#1A1770",
  emerald: "#046C63",
  danger: "#C82828",
  light: "#FFFFFF",
  grey: "#EBEBEB",
  greyDarker: "#D9D9D9",
  greyLighter: "#F2F2F2",
  blueDarker: "#050426",
  success: "#7ABA8E",
  warning: "#F4B286",
  dangerLight: "#F86068",
};