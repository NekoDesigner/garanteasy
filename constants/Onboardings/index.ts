export * from "./Onboardings";
export * from "./@types";
export * from "./screens";
