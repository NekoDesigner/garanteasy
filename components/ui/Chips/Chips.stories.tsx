import React from 'react';
import { Alert, View } from 'react-native';
import { IChipsProps } from './@types';
import Chips from './index';

export default {
  title: 'Chips',
  component: Chips,
};

export const Default = () => (
  <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', padding: 16 }}>
    <Chips label="Default Chips" category="diy" showIcon={false} />
  </View>
);

const chipVariants: IChipsProps[] = [
  { label: 'Électroménage', category: 'hoursehold-electricals', showIcon: true, onPress: () => Alert.alert('Électroménage pressed') },
  { label: 'Petit éléctroménagé', category: 'small-electricals', showIcon: true, onPress: () => Alert.alert('Petit éléctroménagé pressed') },
  { label: 'Bricolage', category: 'diy', showIcon: true, onPress: () => Alert.alert('Bricolage pressed') },
  { label: 'Jardin', category: 'garden', showIcon: true,  onPress: () => Alert.alert('Jardin pressed') },
  { label: 'Mode', category: 'fashion', showIcon: true, onPress: () => Alert.alert('Mode pressed') },
  { label: 'Multimédia', category: 'multimedia', showIcon: true, onPress: () => Alert.alert('Multimédia pressed') },
  { label: 'Autre', category: 'other', showIcon: false, onPress: () => Alert.alert('Autre pressed') },
];

export const WithIcons = () => (
  <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', padding: 16, gap: 24 }}>
    {chipVariants.map((props, idx) => (
      <Chips key={idx} {...props} />
    ))}
  </View>
);