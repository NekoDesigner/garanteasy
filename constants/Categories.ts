import { IChipsProps } from "../components/ui/Chips/@types";

export const CATEGORIES_BASE: IChipsProps[] = [
  { label: 'Électroménagé', category: 'hoursehold-electricals', showIcon: true, onPress: () => {} },
  { label: 'Petit éléctroménagé', category: 'small-electricals', showIcon: true, onPress: () => {} },
  { label: 'Bricolage', category: 'diy', showIcon: true, onPress: () => {} },
  { label: 'Jardin', category: 'garden', showIcon: true,  onPress: () => {} },
  { label: 'Mode', category: 'fashion', showIcon: true, onPress: () => {} },
  { label: 'Multimédia', category: 'multimedia', showIcon: true, onPress: () => {} },
  { label: 'Autre', category: 'other', showIcon: false, onPress: () => {} },
];