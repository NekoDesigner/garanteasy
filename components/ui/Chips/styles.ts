import { StyleSheet } from "react-native";
import { COLORS, SIZES } from "../constants";

const styles = StyleSheet.create({
  button: {
    borderRadius: 100,
    borderWidth: 1,
    backgroundColor: COLORS.emerald,
    paddingVertical: SIZES.padding.xs,
    paddingHorizontal: SIZES.padding.m,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    gap: SIZES.padding.xxs,
  },
  buttonText: {
    color: COLORS.light,
    fontSize: SIZES.font.m,
    fontWeight: SIZES.font.weight.semiBold,
    fontFamily: SIZES.font.familly.default,
  },
});

export default styles;
