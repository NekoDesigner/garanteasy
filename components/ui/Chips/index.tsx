import React from "react";
import { Text, View, TouchableOpacity } from "react-native";
import { COLORS, SIZES } from "../../../constants";
import WashingMachinIcon from "../Icons/WashingMachinIcon";
import { IChipsProps, TChipsIcon } from "./@types";
import styles from "./styles";

const Chips: React.FC<IChipsProps> = ({
  category,
  label,
  size = "s",
  showIcon = true,
  onPress,
}) => {
  const getSize = () => {
    switch (size) {
      case "s":
        return SIZES.icon.s;
      case "xs":
        return SIZES.icon.xs;
      default:
        return SIZES.icon.s;
    }
  };
  const getCategoryIcon = (category: TChipsIcon | React.ReactNode) => {
    switch (category) {
      case "hoursehold-electricals":
      case "small-electricals":
      case "diy":
      case "garden":
      case "fashion":
      case "other":
        return <WashingMachinIcon color={COLORS.light} size={getSize()} />;
      default:
        return category;
    }
  };
  return (
    <TouchableOpacity
      style={styles.button}
      onPress={onPress}
      disabled={onPress === undefined}
    >
      {showIcon && (
        <View testID="category-icon">{getCategoryIcon(category)}</View>
      )}
      <Text style={styles.buttonText}>{label}</Text>
    </TouchableOpacity>
  );
};

export default Chips;
