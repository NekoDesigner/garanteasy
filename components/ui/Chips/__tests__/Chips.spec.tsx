import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Chips from "..";
import { COLORS, SIZES } from "../../constants";

describe("Chips Components", () => {
  const BUTTONS_LIST = [
    {
      category: "hoursehold-electricals",
      label: "Électroménager",
    },
    {
      category: "small-electricals",
      label: "Petits appareils",
    },
    {
      category: "diy",
      label: "Bricolage",
    },
    {
      category: "garden",
      label: "Jardin",
    },
    {
      category: "fashion",
      label: "Mode",
    },
    {
      category: "other",
      label: "Autres",
    },
  ];

  beforeEach(() => {
    jest.clearAllMocks();
  });

  for (const button of BUTTONS_LIST) {
    it(`renders button for category: ${button.category}`, () => {
      const { getByText } = render(
        <Chips category={button.category} label={button.label} />
      );
      expect(getByText(button.label)).toBeTruthy();
    });
  }

  it("renders button with custom icon", () => {
    const customIcon = <div>Custom Icon</div>;
    const { getByText } = render(
      <Chips category={customIcon} label="Custom Category" />
    );
    expect(getByText("Custom Category")).toBeTruthy();
  });

  it("calls onPress when button is pressed", () => {
    const mockOnPress = jest.fn();
    const { getByText } = render(
      <Chips
        category="hoursehold-electricals"
        label="Électroménager"
        onPress={mockOnPress}
      />
    );
    const button = getByText("Électroménager");
    fireEvent.press(button);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it("does not call onPress when button is disabled", () => {
    const mockOnPress = jest.fn();
    const { getByText } = render(
      <Chips
        category="hoursehold-electricals"
        label="Électroménager"
        onPress={undefined}
      />
    );
    const button = getByText("Électroménager");
    fireEvent.press(button);
    expect(mockOnPress).not.toHaveBeenCalled();
  });
  it("renders with small size", () => {
    const { getByTestId } = render(
      <Chips
        category="hoursehold-electricals"
        label="Électroménager"
        size="xs"
      />
    );
    const expected = getByTestId("category-icon").findByProps({
      color: COLORS.light,
    });
    expect(expected).toBeDefined();
    expect(expected.props.size).toBe(SIZES.icon.xs);
  });
  it("renders with default size", () => {
    const { getByTestId } = render(
      <Chips category="hoursehold-electricals" label="Électroménager" />
    );
    const expected = getByTestId("category-icon").findByProps({
      color: COLORS.light,
    });
    expect(expected).toBeDefined();
    expect(expected.props.size).toBe(SIZES.icon.s);
  });
  it("does not show icon when showIcon is false", () => {
    const { queryByTestId } = render(
      <Chips
        category="hoursehold-electricals"
        label="Électroménager"
        showIcon={false}
      />
    );
    const icon = queryByTestId("category-icon");
    expect(icon).toBeNull();
  });
});
