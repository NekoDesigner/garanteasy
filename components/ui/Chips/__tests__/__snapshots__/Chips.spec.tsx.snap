// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chips Components renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#046C63",
      "borderRadius": 100,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "gap": 4,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    testID="category-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={62}
      vbWidth={60}
      width={20}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M57 53.705a7.364 7.364 0 01-7.364 7.363H10.364A7.364 7.364 0 013 53.705V18.114h54v35.59zM30 23.023c-8.134 0-14.727 6.593-14.727 14.727S21.866 52.477 30 52.477s14.727-6.593 14.727-14.727S38.134 23.023 30 23.023zm0 1.84c7.117 0 12.886 5.77 12.886 12.887 0 7.117-5.77 12.886-12.886 12.886-7.117 0-12.886-5.77-12.886-12.886 0-7.117 5.77-12.886 12.886-12.886zM49.636.932A7.364 7.364 0 0157 8.295v7.364H3V8.295A7.364 7.364 0 0110.364.932h39.272zM36.75 9.522a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682H36.75zm9.818 0a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682h-3.682z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Ubuntu",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Électroménager
  </Text>
</View>
`;

exports[`Chips Components renders correctly 2`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#046C63",
      "borderRadius": 100,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "gap": 4,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    testID="category-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={62}
      vbWidth={60}
      width={20}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M57 53.705a7.364 7.364 0 01-7.364 7.363H10.364A7.364 7.364 0 013 53.705V18.114h54v35.59zM30 23.023c-8.134 0-14.727 6.593-14.727 14.727S21.866 52.477 30 52.477s14.727-6.593 14.727-14.727S38.134 23.023 30 23.023zm0 1.84c7.117 0 12.886 5.77 12.886 12.887 0 7.117-5.77 12.886-12.886 12.886-7.117 0-12.886-5.77-12.886-12.886 0-7.117 5.77-12.886 12.886-12.886zM49.636.932A7.364 7.364 0 0157 8.295v7.364H3V8.295A7.364 7.364 0 0110.364.932h39.272zM36.75 9.522a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682H36.75zm9.818 0a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682h-3.682z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Ubuntu",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Petits appareils
  </Text>
</View>
`;

exports[`Chips Components renders correctly 3`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#046C63",
      "borderRadius": 100,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "gap": 4,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    testID="category-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={62}
      vbWidth={60}
      width={20}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M57 53.705a7.364 7.364 0 01-7.364 7.363H10.364A7.364 7.364 0 013 53.705V18.114h54v35.59zM30 23.023c-8.134 0-14.727 6.593-14.727 14.727S21.866 52.477 30 52.477s14.727-6.593 14.727-14.727S38.134 23.023 30 23.023zm0 1.84c7.117 0 12.886 5.77 12.886 12.887 0 7.117-5.77 12.886-12.886 12.886-7.117 0-12.886-5.77-12.886-12.886 0-7.117 5.77-12.886 12.886-12.886zM49.636.932A7.364 7.364 0 0157 8.295v7.364H3V8.295A7.364 7.364 0 0110.364.932h39.272zM36.75 9.522a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682H36.75zm9.818 0a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682h-3.682z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Ubuntu",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Bricolage
  </Text>
</View>
`;

exports[`Chips Components renders correctly 4`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#046C63",
      "borderRadius": 100,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "gap": 4,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    testID="category-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={62}
      vbWidth={60}
      width={20}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M57 53.705a7.364 7.364 0 01-7.364 7.363H10.364A7.364 7.364 0 013 53.705V18.114h54v35.59zM30 23.023c-8.134 0-14.727 6.593-14.727 14.727S21.866 52.477 30 52.477s14.727-6.593 14.727-14.727S38.134 23.023 30 23.023zm0 1.84c7.117 0 12.886 5.77 12.886 12.887 0 7.117-5.77 12.886-12.886 12.886-7.117 0-12.886-5.77-12.886-12.886 0-7.117 5.77-12.886 12.886-12.886zM49.636.932A7.364 7.364 0 0157 8.295v7.364H3V8.295A7.364 7.364 0 0110.364.932h39.272zM36.75 9.522a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682H36.75zm9.818 0a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682h-3.682z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Ubuntu",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Jardin
  </Text>
</View>
`;

exports[`Chips Components renders correctly 5`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#046C63",
      "borderRadius": 100,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "gap": 4,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    testID="category-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={62}
      vbWidth={60}
      width={20}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M57 53.705a7.364 7.364 0 01-7.364 7.363H10.364A7.364 7.364 0 013 53.705V18.114h54v35.59zM30 23.023c-8.134 0-14.727 6.593-14.727 14.727S21.866 52.477 30 52.477s14.727-6.593 14.727-14.727S38.134 23.023 30 23.023zm0 1.84c7.117 0 12.886 5.77 12.886 12.887 0 7.117-5.77 12.886-12.886 12.886-7.117 0-12.886-5.77-12.886-12.886 0-7.117 5.77-12.886 12.886-12.886zM49.636.932A7.364 7.364 0 0157 8.295v7.364H3V8.295A7.364 7.364 0 0110.364.932h39.272zM36.75 9.522a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682H36.75zm9.818 0a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682h-3.682z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Ubuntu",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Mode
  </Text>
</View>
`;

exports[`Chips Components renders correctly 6`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#046C63",
      "borderRadius": 100,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "gap": 4,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <View
    testID="category-icon"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={62}
      vbWidth={60}
      width={20}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M57 53.705a7.364 7.364 0 01-7.364 7.363H10.364A7.364 7.364 0 013 53.705V18.114h54v35.59zM30 23.023c-8.134 0-14.727 6.593-14.727 14.727S21.866 52.477 30 52.477s14.727-6.593 14.727-14.727S38.134 23.023 30 23.023zm0 1.84c7.117 0 12.886 5.77 12.886 12.887 0 7.117-5.77 12.886-12.886 12.886-7.117 0-12.886-5.77-12.886-12.886 0-7.117 5.77-12.886 12.886-12.886zM49.636.932A7.364 7.364 0 0157 8.295v7.364H3V8.295A7.364 7.364 0 0110.364.932h39.272zM36.75 9.522a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682H36.75zm9.818 0a1.84 1.84 0 100 3.683h3.682a1.841 1.841 0 000-3.682h-3.682z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "Ubuntu",
        "fontSize": 16,
        "fontWeight": "700",
      }
    }
  >
    Autres
  </Text>
</View>
`;
